src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -968,6 +978,118 @@ class spell_dru_t4_2p_bonus : public SpellScriptLoader
         }
 };
 
+// 783 - Travel Form
+class spell_dru_travel_form : public SpellScriptLoader
+{
+public:
+    spell_dru_travel_form() : SpellScriptLoader("spell_dru_travel_form") { }
+
+    class spell_dru_travel_form_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dru_travel_form_SpellScript);
+
+        bool Load() override
+        {
+            return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+        }
+
+        SpellCastResult CheckCast()
+        {
+            Player* player = GetCaster()->ToPlayer();
+            if (player->IsInDisallowedMountForm())
+                return SPELL_FAILED_NOT_SHAPESHIFT;
+            if (!player->IsInWater() && !player->GetMap()->IsOutdoors(player->GetPositionX(), player->GetPositionY(), player->GetPositionZ()))
+                return SPELL_FAILED_ONLY_OUTDOORS;
+
+            return SPELL_CAST_OK;
+        }
+
+        void Register() override
+        {
+            OnCheckCast += SpellCheckCastFn(spell_dru_travel_form_SpellScript::CheckCast);
+        }
+    };
+
+    class spell_dru_travel_form_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_dru_travel_form_AuraScript);
+
+        void AfterApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            Player* player = GetTarget()->ToPlayer();
+            if (player->IsInWater())
+                player->CastSpell(player, DRUID_AQUATIC_FORM, true);
+            else if (player->IsInCombat())
+                player->CastSpell(player, DRUID_STAG_FORM, true);
+            else if ((player->GetMapId() == 530) || (player->GetMapId() == 571 && player->HasSpell(54197)) || ((player->GetMapId() == 0 || player->GetMapId() == 1) && player->HasSpell(90267)))
```

Comment:
I'm not sure, I've looked at player->CanFlyInZone(), and it seems outdated, but I could be wrong.